tests/unit/type/string.tcl

```diff
@@ -582,6 +582,59 @@ if {[string match {*jemalloc*} [s mem_allocator]]} {
       set err1
     } {*WRONGTYPE*}
 
+    test "SET with IFEQ conditional" {
+        r del foo
+
+        r set foo "initial_value"
+
+        assert_equal {OK} [r set foo "new_value" ifeq "initial_value"]
+        assert_equal "new_value" [r get foo]
+
+        assert_equal {} [r set foo "should_not_set" ifeq "wrong_value"]
```

Comment:
@valkey-io/core-team On failure of compare/set instead of nil value we should return an error with old value in it. Otherwise, a client would need to perform another GET operation.

Reply:
return an error with old value seems odd to me. i think in normal cases, a client is unlikely get an error in set IFEQ, they hold the old value in some caes, and if the old value is unvalid, this mean client should abort the set, this usually means the client should abort the entire business logic. In this case, client should GET the new value as needed and usually they don't really need the new value, they juse want the result of whether the SET succeeded or not.

Reply:
@enjoy-binbin I understand if the value is updated from a DB to Valkey as a cache, the client won't benefit much from the value present. 

However, if I think Valkey being used as a datastore and value is updated by multiple clients based on the value stored (let say increment old value by 1). for this case, they would need to know the previous value.

Let's hear other devs opinion on this and resolve this.

Reply:
I like the idea of returning value if the IFEQ fails, not as an error, just as a string. That way you can also check to see if someone else did the same work you did (and silently succeed) or retry you work again.

Reply:
With the current implementation of `SET key value IFEQ compare GET` in this PR, the old value is returned regardless of whether it's matching the `compare` string or not. This can be used for the use case @madolson described.

Without GET, I don't think we can return the old value on mismatch, because of this corner case: When the old value is the string "OK", if the client library returns "OK" as a string (i.e. if it doesn't distinguish between bulk string and simple string), the user can't tell if "OK" means that the SET succeeded or if just returned the old value which was the string "OK".